src/main/java/com/alvarium/annotators/AnnotatorInfo.java

```diff
@@ -0,0 +1,57 @@
+
+/*******************************************************************************
+ * Copyright 2023 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+package com.alvarium.annotators;
+
+import java.io.Serializable;
+
+import com.alvarium.contracts.AnnotationType;
+import com.alvarium.serializers.AnnotatorInfoConverter;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+public class AnnotatorInfo implements Serializable {
+    private final AnnotationType kind;
+    private final Object config;
+
+    public AnnotatorInfo(AnnotationType kind) {
+        this.kind = kind;
+        this.config = new Object();
```

Comment:
Something to think about -- Declaring a property as `Object` is generally a bad practice b/c who the hell knows what it is? Sure, the property name is `config` but that's meaningless b/c the type isn't constrained.

Suggest you look at and consider the approach taken in the Go SDK which defines an interface named `Configuration` with a single function `AsString()`. This would at least give some constraints on what the property can be set to in that the value has to be a type that implements an `AsString()` method. Plus the interface name is explicit.

I realize what's happening here is a desire to provide flexibility for an arbitrary config for each annotator.

Further, as to having a separate property for `kind` -- Is that because you're thinking each Annotator might have a unique config of a separate type? In other words, we'd want to validate that the PkiAnnotatorConfig wasn't supplied for the SourceAnnotator? I'm not 100% sold on that yet, and we might want wait for our use cases to develop a bit more. `kind` might actually be a property on a generalized annotator config until we start to learn more about specialization -- especially since right now all annotators but one will have no extra config info.

Reply:
I'm not big on declaring `Object` properties as well, but I didn't want to break the pattern used previously in both the Java and Go SDKs. We're doing the same thing for stream provider configuration (see [#3](https://github.com/project-alvarium/alvarium-sdk-java/pull/39#discussion_r703003467)) but if that may be problematic I can open an issue to treat it the same way you described. I also see that the Go SDK [does have a `Configuration` interface](https://github.com/project-alvarium/alvarium-sdk-go/blob/main/pkg/config/interface.go) but I couldn't find it being used anywhere, instead a [generic `interface{}` was used](https://github.com/project-alvarium/alvarium-sdk-go/blob/0532be91d69b6eadeee9397d327244bc0b31ef8a/pkg/config/stream.go#L28) in its place which was probably not intentional... should I open an issue for that as well ?

As for the `kind` property, that's exactly what I was going for. I'll go for a flatter config structure like this one

```json
{
    annotators: [
        {
            "kind": "tpm"
        },
        {
            "kind": "mock",
            "shouldSatisfy": false
        },
        {
            "kind": "vulnerability",
            "provider": {
                 "host": "example.com",
                 "port": 80,
                 "protocol": "https"
             }
        },
    ]
}
```

Reply:
The PR size in 1bdccdc got somewhat larger after rebasing, the vulnerability annotator changes played a part in that so if its too large let me know and I'll move these changes to another PR. 

I've tried to flatten the configuration, avoid using `Object` types, and created an interface for the annotator config. One thing I wasn't able to address as well as you probably wanted is holding off on unique configs for each annotator. 

I also moved the `VulnerabilityAnnotatorConfig` to the `annotators.vulnerability` package because one of the factories depended on it and I did not want it importing something from the `annotators` package. Not sure if this is the best approach at the moment.

My thought process probably got a bit mixed up by the end of the work here but I thought I'd push some of these changes for now till I give others more thought, specifically the vulnerability config fields might need some restructuring + the other stuff I mentioned.

Reply:
Changes made in 172ec39
- Removed vulnerability annotator changes to reduce PR size
- Changed the annotator config from an interface to a regular class that can be used instead of the `DefaultAnnotatorConfig` to reduce unnecessary number of files.